src/main/tls_listen.c

```diff
@@ -375,7 +375,6 @@ static int tls_socket_recv(rad_listen_t *listener)
 	if (listener->proxy_protocol) {
 		rcode = proxy_protocol_check(listener, request);
 		if (rcode < 0) goto do_close;
-		if (rcode == 0) return 1;
```

Comment:
The `proxy_protocol_check()` function should return:

* -1 on error
* 0 for "need to read more data"
* 1 for "done

I'll push a slightly different fix.

Reply:
That works.